relay-filter/src/config.rs

```diff
@@ -416,34 +409,60 @@ pub struct GenericFiltersConfig {
     ///
     /// The map contains unique filters, meaning there are no two filters with
     /// the same id. See struct docs for more details.
-    #[serde(
-        default,
-        skip_serializing_if = "IndexMap::is_empty",
-        with = "generic_filters_custom_serialization"
-    )]
-    pub filters: IndexMap<String, GenericFilterConfig>,
+    #[serde(default, skip_serializing_if = "GenericFiltersMap::is_empty")]
+    pub filters: GenericFiltersMap,
 }
 
 impl GenericFiltersConfig {
     /// Returns true if the filters are not declared.
     pub fn is_empty(&self) -> bool {
-        self.filters.is_empty()
+        self.filters.0.is_empty()
     }
 }
 
-mod generic_filters_custom_serialization {
-    use super::*;
+/// Map of generic filters, mapping from the id to the filter itself.
+#[derive(Clone, Debug, Default)]
+pub struct GenericFiltersMap(IndexMap<String, GenericFilterConfig>);
+
+impl GenericFiltersMap {
+    /// Returns an empty map.
+    pub fn new() -> Self {
+        GenericFiltersMap(IndexMap::new())
+    }
+
+    /// Returns whether the map is empty.
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+impl From<Vec<GenericFilterConfig>> for GenericFiltersMap {
+    fn from(filters: Vec<GenericFilterConfig>) -> Self {
+        let mut map = IndexMap::with_capacity(filters.len());
+        for filter in filters {
+            map.insert(filter.id.clone(), filter.clone());
```

Comment:
You need to check if there is not already an item in the map, same thing you do in the Deserialize, order is important.